integrations/gradle-plugin/src/main/groovy/org/robolectric/gradleplugin/DownloadAndroidSdks.groovy

```diff
@@ -0,0 +1,101 @@
+package org.robolectric.gradleplugin
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.tasks.TaskAction
+
+class DownloadAndroidSdks extends DefaultTask {
+
+  @TaskAction
+  void run() {
+    GradlePlugin.Config config = getProject().getExtensions().getByType(GradlePlugin.Config.class);
+
+    def defaultSdks = loadPropertiesResourceFile("org.robolectric.GradlePlugin.sdks.properties")
+    def enabledSdks = figureSdks(config.sdks, defaultSdks)
+    Properties sdkDeps = new Properties()
+    enabledSdks.keySet().forEach { Integer apiLevel ->
+      def sdkCoords = enabledSdks.get(apiLevel)
+      if (sdkCoords instanceof File) {
+        sdkCoords = project.files(sdkCoords)
+      }
+      def sdkCfg = project.configurations.create("sdk$apiLevel")
+      project.dependencies.add("sdk$apiLevel", sdkCoords)
+
+      def sdkFiles = sdkCfg.resolve()
+      if (sdkFiles.size() != 1) {
+        throw new IllegalStateException("weird, $sdkCoords returned $sdkFiles, not one file")
+      }
+      sdkDeps[apiLevel.toString()] = sdkFiles[0].toString()
+    }
+
+    def outDir = getGeneratedDir(project)
+    outDir.mkdirs()
+    def outFile = new File(outDir, 'org.robolectric.sdks.properties')
+    def out = outFile.newOutputStream()
+    sdkDeps.store(out, null)
+    out.close()
+  }
+
+  static File getGeneratedDir(Project project) {
+    return new File(project.buildDir, "generated/robolectric")
+  }
+
+  static void addGeneratedResourcesDirToTestSourceSets(Project project) {
+    def outDir = getGeneratedDir(project)
+
+    project.android.sourceSets.forEach { sourceSet ->
+      if (sourceSet.name.startsWith("test")) {
+        sourceSet.resources.srcDir(outDir)
+      }
+    }
+  }
+
+  static Map<Integer, Object> figureSdks(Object configSdks, Properties defaultSdks) {
+    def map = new HashMap<Integer, Object>()
+
+    def add = { Integer apiLevel ->
+      def coordinates = defaultSdks.getProperty(apiLevel.toString())
+      if (coordinates == null) {
+        throw new IllegalArgumentException("Unknown API level $apiLevel")
+      }
+      map.put(apiLevel, coordinates)
+    }
+
+    if (configSdks == null) {
+      defaultSdks.forEach { k, v ->
```

Comment:
Nit: Might be useful to log this via `logger.debug` to Gradle that the SDKs (all?) from `org.robolectric.GradlePlugin.sdks.properties` are loaded here.